custom_components/tahoma/light.py

```diff
@@ -68,29 +85,44 @@ def supported_features(self) -> int:
         if "wink" in self.tahoma_device.command_definitions:
             supported_features |= SUPPORT_EFFECT
 
+        if "setRGB" in self.tahoma_device.command_definitions:
+            supported_features |= SUPPORT_COLOR
+
         return supported_features
 
+    def _apply_action(self, cmd_name, *args):
+        """Apply an action and wait for it to complete."""
+        exec_id = self.apply_action(cmd_name, *args)
+        while exec_id in self.controller.get_current_executions():
+            continue
+
     def turn_on(self, **kwargs) -> None:
         """Turn the light on."""
         self._state = True
-        self._skip_update = True
+
+        _LOGGER.warning(f"light.turn_on kwargs: {kwargs}")
+
+        if ATTR_HS_COLOR in kwargs:
+            rgb = color_util.color_hs_to_RGB(*kwargs[ATTR_HS_COLOR])
+            self._rgb = [int(float(c)) for c in rgb]
```

Comment:
Can we already set `self._rgb` here? What if the command fails, in that case the state is incorrect? 

Or should we implement this behavior in all integrations, where we directly update the state, but change it back if the command fails. (which happens via the update() function)

Reply:
This is one still applies, but is now renamed to `self._hs_color`

Reply:
Agreed, `self._hs_color` should be set in the update triggered after the turn_on, now that the command awaits for its completion.

Reply:
isn't that the same for `_brightness` and `_effect` though?

Reply:
You are right, let's keep it for now to have the same behavior across this integration.

Reply:
ok, let me push that and we can merge.